gateway-ha/src/main/java/io/trino/gateway/baseapp/BaseApp.java

```diff
@@ -115,8 +119,9 @@ public void initialize(Bootstrap<T> bootstrap)
     public void run(T configuration, Environment environment)
             throws Exception
     {
+        ((DefaultServerFactory) configuration.getServerFactory()).setRequestLogFactory(new GatewayRequestLogFactory());
```

Comment:
You could avoid the cast by overriding `getServerFactory` in `AppConfiguration` to return a `DefaultServerFactory` or `AbstractServerFactory` and `setServerFactory` to ensure only a `DefaultServerFactory`/`AbstractServerFactory` can be set.